### DISCUSSION TITLE
How to parse whitespace sensitive grammar(Python-like) by textX？

### DISCUSSION THREAD

**Discussion by tongorz** [textX/textX#327] [Q&A]
How to parse whitespace sensitive grammar(Python-like) by textX？

**Comment by Aluriak** [comment]
I tried to achieve this few weeks ago, and it seems currently not possible.
My problem was to parse identifiers groups that are presented by line :

    A B C
    A D

Problem was that newlines are discarded/replaced by TextX, avoiding me to retrieve the two entities separatly (instead of a single one with 5 fields) without modifying the language to be parsed.
If you look at all examples presented in TextX doc, they all avoid this problem by having starting/closing token (which is not `\n`).

This problem is at least known since #15.

**Comment by igordejanovic** [comment]
Right. There is no direct support for whitespace sensitive languages at the moment. #15 as @Aluriak noted is going to deal with this.

@Aluriak Your problem is different. What you can do is to disable whitespace skipping using `ws` [parser parameter](http://www.igordejanovic.net/textX/parser_config/#whitespace-handling) to redefine whitespaces or disable whitespace entirely using `skipws` param. You can also do that [per rule](http://www.igordejanovic.net/textX/grammar/#rule-modifiers). This way you can catch whitespaces in your grammar. For your particular case, when you should match inside single line only, there is even simpler solution by using [`eolterm` repetition modifier](http://www.igordejanovic.net/textX/grammar/#repetition-modifiers).

See for example implementation of [pyFlies conditions](https://github.com/igordejanovic/pyFlies/blob/d4689139c25120e40b41bb276a41367399d266c9/examples/Simon/Simon.pf#L13). In this DSL each state is a separate line.

**Comment by Aluriak** [comment]
Ok, i retried, using the information you provides. It works now :)
Thank you a lot !

**Comment by Halbeard** [comment]
Is there any update on this issue?
I'm writing a markdown like DSL that is intended to be user extendable and this is basically the only thing that forces me to choose arpeggio over textx.

**Comment by igordejanovic** [comment]
Hi @Halbeard No, unfortunately, there is no progress on this issue at the moment.
